kernel/src/vm/vmar/vm_mapping.rs

```diff
@@ -300,9 +322,22 @@ impl VmMapping {
         );
 
         let vm_perms = self.perms - VmPerms::WRITE;
+        let mut rss_increment: usize = 0;
+
         'retry: loop {
             let preempt_guard = disable_preempt();
-            let mut cursor = vm_space.cursor_mut(&preempt_guard, &(start_addr..end_addr))?;
+
+            let mut cursor = match vm_space.cursor_mut(&preempt_guard, &(start_addr..end_addr)) {
+                Ok(cursor) => cursor,
+                Err(e) => {
+                    return (rss_increment, Err(e.into()));
+                }
+            };
+
+            let mut add_rss = |count| {
+                rss_increment += count;
+            };
+
```

Comment:
- Remove this blank line as the only purpose of `add_rss` is to be captured by the following closure.
 - I don't think `add_rss` itself needs to be a closure, how about:
```rust
            let ref_to_rss_increment = &mut rss_increment;
            let operate =
                move |commit_fn: &mut dyn FnMut()
                    -> core::result::Result<UFrame, VmoCommitError>| {
                    // ..
                    *ref_to_rss_increment += 1;
```